packages/react-instantsearch-hooks-dom/src/ui/SearchBox.tsx

```diff
@@ -56,20 +56,25 @@ export type SearchBoxTranslations = {
   resetTitle: string;
 };
 
-export type SearchBoxProps = React.HTMLAttributes<HTMLDivElement> &
-  Pick<React.HTMLAttributes<HTMLFormElement>, 'onSubmit'> &
-  Required<Pick<React.HTMLAttributes<HTMLFormElement>, 'onReset'>> &
-  Pick<
-    React.InputHTMLAttributes<HTMLInputElement>,
-    'placeholder' | 'onChange'
-  > & {
-    inputRef: React.RefObject<HTMLInputElement>;
-    isSearchStalled: boolean;
-    value: string;
+type WrapperProps = Omit<React.ComponentProps<'div'>, 'onChange' | 'onReset'>;
+
+export type SearchBoxWidgetProps = WrapperProps &
+  Pick<React.ComponentProps<'form'>, 'onSubmit'> &
+  Pick<React.ComponentProps<'input'>, 'placeholder'> & {
     resetIconComponent?: React.JSXElementConstructor<IconProps>;
     submitIconComponent?: React.JSXElementConstructor<IconProps>;
     loadingIconComponent?: React.JSXElementConstructor<IconProps>;
     classNames?: Partial<SearchBoxClassNames>;
+  };
+
+export type SearchBoxProps = SearchBoxWidgetProps &
+  Required<
+    Pick<React.ComponentProps<'form'>, 'onReset'> &
+      Pick<React.ComponentProps<'input'>, 'onChange'>
+  > & {
```

Comment:
as these overlap, is the end result required or optional? are we able to use a similar pattern for infinite hits?

Reply:
They are required and omitted from `WrapperProps` with which they overlapped.

Reply:
there's no omitting happening in this one, in InfiniteHits/Hits there is

Reply:
The definition on L59 seems to be working though:

```ts
type WrapperProps = Omit<React.ComponentProps<'div'>, 'onChange' | 'onReset'>;
```

The only overlap I see is with `onSubmit` which I'll fix but the other one are correctly marked as required from `SearchBoxProps`. Am I missing something?

Reply:
🤦 I didn't see that line! It seems good to go with for now then, but maybe with override keys or a similar utility it would be simpler to read?

https://github.com/algolia/react-instantsearch/blob/51e8818fce224819230c8bf6dea2a08d71d9be29/test/mock/createSearchClient.ts#L19-L21